src/lib/src/component/navigation/navigation.component.spec.ts

```diff
@@ -515,4 +521,209 @@ describe('NavigationComponent', () => {
     flush();
   }));
 
+
+  it('sets expanded state when right arrow key is pressed', () => {
+    // given
+    setupWorkspace(component, fixture);
+    let element = component.workspace.getElement('subfolder');
+    component.uiState.selectedElement = element;
+    component.uiState.setExpanded(element.path, false);
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_RIGHT});
+
+    // then
+    let expandedState = component.uiState.isExpanded(element.path);
+    expect(expandedState).toBeTruthy();
+  });
+
+  it('keeps expanded state when right arrow key is pressed', () => {
+    // given
+    setupWorkspace(component, fixture);
+    let element = component.workspace.getElement('subfolder');
+    component.uiState.selectedElement = element;
+    component.uiState.setExpanded(element.path, true);
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_RIGHT});
+
+    // then
+    let expandedState = component.uiState.isExpanded(element.path);
+    expect(expandedState).toBeTruthy();
+  });
+
+  it('sets collapsed state when left arrow key is pressed', () => {
+    // given
+    setupWorkspace(component, fixture);
+    let element = component.workspace.getElement('subfolder');
+    component.uiState.selectedElement = element;
+    component.uiState.setExpanded(element.path, true);
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_LEFT});
+
+    // then
+    let expandedState = component.uiState.isExpanded(element.path);
+    expect(expandedState).toBeFalsy();
+  });
+
+  it('keeps collapsed state when left arrow key is pressed', () => {
+    // given
+    setupWorkspace(component, fixture);
+    let element = component.workspace.getElement('subfolder');
+    component.uiState.selectedElement = element;
+    component.uiState.setExpanded(element.path, false);
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_LEFT});
+
+    // then
+    let expandedState = component.uiState.isExpanded(element.path);
+    expect(expandedState).toBeFalsy();
+  });
+
+  it('selects the next sibling element when the down arrow key is pressed', async(() => {
+    // given
+    setupWorkspace(component, fixture);
+    component.uiState.selectedElement = nonExecutableFile;
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_DOWN});
```

Comment:
naming is a bit confusing here 'keyup' with key: KEY_DOWN ? maybe renaming to NAVIGATE_NEXT